SDWebImage/Core/SDImageCacheDefine.m

```diff
@@ -128,12 +126,6 @@ void SDSetDecodeOptionsToContext(SDWebImageMutableContext * _Nonnull mutableCont
         if (lazyDecode) {
             // lazyDecode = NO means we should not forceDecode, highest priority
             shouldDecode = NO;
-        } else if ([image.class conformsToProtocol:@protocol(SDAnimatedImage)]) {
```

Comment:
Dupliacted check, the logic is inside `+[SDImageCoderHelper shouldDecodeImage:]`